assets/js/atomic/components/product/button/index.js

```diff
@@ -28,51 +26,34 @@ import { decodeEntities } from '@wordpress/html-entities';
  * @param {number} productId  The product id for the product connection to the
  *                            cart.
  *
- * @return {Object} Returns an object with the following properties:
- *    @type {number}   cartQuantity  The quantity of the product currently in
- *                                   the cart.
- *    @type {boolean}  addingToCart  Whether the product is currently being
- *                                   added to the cart (true).
- *    @type {boolean}  cartIsLoading Whether the cart is being loaded.
- *    @type {Function} addToCart     An action dispatcher for adding a single
- *                                   quantity of the product to the cart.
- *                                   Receives no arguments, it operates on the
- *                                   current product.
+ * @return {StoreCartItemAddToCart} An object exposing data and actions relating
+ *                                  to add to cart functionality.
  */
 const useAddToCart = ( productId ) => {
```

Comment:
I was secretly hoping we could use an existing hook as part of fixing this – unfortunate having a custom hook just for this add-to-cart button.

Reply:
Nice that this now leverages `useStoreCart` and has typedef; perhaps we can move this out of product button soon.

Reply:
Done and done :)